### PR TITLE
feat(typeahead): add uib- prefix

### PR DESCRIPTION
**Pull request by BobbieBarker**

### CODE REVIEW

File: src/typeahead/test/typeahead-highlight.spec.js [hunk 1]
```diff
@@ -49,3 +49,30 @@ describe('typeaheadHighlight', function () {
     expect(logSpy).toHaveBeenCalled();
   });
 });
+
```
Comment (wesleycho) [inline src/typeahead/test/typeahead-highlight.spec.js]: Add a comment about these being deprecation tests

File: src/typeahead/test/typeahead-popup.spec.js [hunk 2]
```diff
@@ -55,3 +55,45 @@ describe('typeaheadPopup - result rendering', function() {
     expect($rootScope.select).toHaveBeenCalledWith(2);
   });
 });
+
```
Comment (wesleycho) [inline src/typeahead/test/typeahead-popup.spec.js]: Ditto

File: src/typeahead/test/typeahead.spec.js [hunk 3]
```diff
@@ -1007,10 +1007,56 @@ describe('typeahead tests', function() {
...
     it('should open typeahead if input is changed to empty string if defined threshold is 0', function() {
-      var element = prepareInputEl('<div><input ng-model="result" typeahead="item for item in source | filter:$viewValue" typeahead-min-length="0"></div>');
+      var element = prepareInputEl('<div><input ng-model="result" uib-typeahead="item for item in source | filter:$viewValue" typeahead-min-length="0"></div>');
       changeInputValueTo(element, '');
-
       expect(element).toBeOpenWithActive(3, 0);
     });
   });
 });
+
```
Comment (wesleycho) [inline src/typeahead/test/typeahead.spec.js]: Ditto

File: src/typeahead/typeahead.js [hunk 4]
```diff
@@ -498,7 +497,522 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.position'])
...
+    function($compile, $parse, $q, $timeout, $document, $window, $rootScope, $position, typeaheadParser, $log, $typeaheadSuppressWarning) {
+    var HOT_KEYS = [9, 13, 27, 38, 40];
+    var eventDebounceTime = 200;
+    return {
+      require: ['ngModel', '^?ngModelOptions'],
+      link: function(originalScope, element, attrs, ctrls) {
+        if (!$typeaheadSuppressWarning) {
+          $log.warn('typeahead is now deprecated. Use uib-typeahead instead.');
+        }
+        var modelCtrl = ctrls[0];
```
Comment (wesleycho) [inline src/typeahead/typeahead.js]: This is a huge amount of duplication...@foxandxss, do we want a directive controller for this before doing a release, or maybe slate a refactor for 0.14.1 and let this temporarily slide?

File: src/typeahead/typeahead.js [hunk 5]
```diff
@@ -523,3 +1040,5 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.position'])
       return matchItem;
     };
   }]);
+  
+
```
Comment (wesleycho) [inline src/typeahead/typeahead.js]: Delete this

File: src/typeahead/typeahead.js [hunk 6]
```diff
@@ -498,7 +497,522 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.position'])
...
+        position: '&',
+        moveInProgress: '=',
+        select: '&'
+      },
+      replace: true,
+      templateUrl: function(element, attrs) {
+        return attrs.popupTemplateUrl || 'template/typeahead/typeahead-popup.html';
+      },
+      link: function(scope, element, attrs) {
+
```
Comment (wesleycho) [inline src/typeahead/typeahead.js]: Remove this line.

File: src/typeahead/typeahead.js [hunk 7]
```diff
@@ -498,7 +497,522 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.position'])
...
+        if (appendToBody) {
+          $document.find('body').append($popup);
+        } else if (appendToElementId !== false) {
+          angular.element($document[0].getElementById(appendToElementId)).append($popup);
+        } else {
+          element.after($popup);
+        }
+      }
+    };
+
```
Comment (wesleycho) [inline src/typeahead/typeahead.js]: Remove this

File: src/typeahead/typeahead.js (hunk 6 above)
Comment (Foxandxss) [inline src/typeahead/typeahead.js]: What line? it is not clear, I Guess you mean 954.

Comment (wesleycho) [inline src/typeahead/typeahead.js]: I meant the line this comment is - it's unnecessary whitespace

Comment (Foxandxss) [inline src/typeahead/typeahead.js]: Oh, right, was a bit misleading that whitespace :P

**Comment by wesleycho** [comment]
Going to merge this with code style fixes, but I am slating refactoring the typeahead into a controller for 0.14.1 to remove the duplication
